### ISSUE TITLE
XBL components: native support for itemsets

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#768]
- add an `itemset` mode
- itemset can be provided _from the outside_, like for built-in selection controls
- itemset can also be provided _from the inside_
  - tentatively, this would be placed within a local model, requiring #1520
    - if so, need way to point to itemset by id (`xxbl:itemsetid="my-itemset"`?)
  - could also be placed directly under `xbl:binding` possibly, like handlers
    - if so, no need to refer by id
- `xxf:itemset()` must then work when applied to the binding

Historical: see also [Thoughts about XBL.next](http://wiki.orbeon.com/forms/projects/xbl/toughts-about-xbl-next).

**Comment by avernet** [comment]
[+1 from community](http://stackoverflow.com/questions/31430654/in-orbeon-xform-how-to-get-selected-drop-down-label-on-change-of-drop-down-in-a)

**Comment by ebruchez** [comment]
Note that since Orbeon Forms 4.8 (and 4.7.1), `xxf:itemset()` works with some XBL components at least, see #1856. And so it should work with `databound-select1.xbl` as well.

**Comment by avernet** [comment]
@ebruchez What is the status on this one? Does `xxf:itemset()` indeed work with some XBL components, as stated in the previous comment? What is the mechanism? I can't find documentation for this either in our XBL doc or the doc for `xxf:itemset()`.

**Comment by ebruchez** [comment]
@avernet Yes it works with some components, namely if the target control is an XBL component, it looks for the first nested control with an itemset and returns that itemset.

**Comment by avernet** [comment]
@ebruchez OK. Should we document this behavior, and close this issue? Or do you have something else in mind?

**Comment by ebruchez** [comment]
Since it was kind of a hack, I am not sure we should document it. Or if we do, maybe document as "works with such and such specific components too".

**Comment by ebruchez** [comment]
We do have a `selection` (not `itemset`) mode since Orbeon Forms 2018.1 (#2428). This only marks the control but doesn't add any functionality at this point.

**Comment by ebruchez** [comment]
See also #4042.

**Comment by avernet** [comment]
[+1 from user](https://stackoverflow.com/questions/62530440)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/7545500/messages/2869622444)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/3237397/messages/6739988019)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/36258888/messages/7704225640)

**Comment by avernet** [comment]
The heuristic in `findSelectionControl`, which involves looking for the first `XFormsSelect1Control` in the component, doesn't work for `databound-select1-search.xbl` because it doesn't contain an `xf:select1`.

While we would prefer a more general solution, we discussed with @ebruchez the possibility of having `XXFormsItemset.scala` implement something specific to `databound-select1-search.xbl`. Specifically, we would need to examine the content of the `result-json` instance and create an `Itemset` based on that content.

**Comment by avernet** [comment]
I created #6444 for the specific case of the Dynamic dropdown with search (`databound-select1-search.xbl`).
